src/Files.App/Shell/Win32API.cs

```diff
@@ -786,28 +786,17 @@ public static void OpenFolderInExistingShellWindow(string folderPath)
         public static extern int SHQueryRecycleBin(string pszRootPath,
             ref SHQUERYRBINFO pSHQueryRBInfo);
 
-        public static bool InstallInf(string filePath)
+        public static async Task InstallInf(string filePath)
         {
-            try
-            {
-                using Process process = new Process();
-                process.StartInfo.FileName = "InfDefaultInstall.exe";
-                process.StartInfo.Verb = "runas";
-                process.StartInfo.UseShellExecute = true;
-                process.StartInfo.CreateNoWindow = true;
-                process.StartInfo.Arguments = $"{filePath}";
-                process.Start();
-                if (process.WaitForExit(30 * 1000))
-                {
-                    return process.ExitCode == 0;
-                }
-                return false;
-            }
-            catch (Win32Exception)
-            {
-                // If user cancels UAC
-                return false;
-            }
+            var cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(30 * 1000));
+            using Process process = new Process();
+            process.StartInfo.FileName = "InfDefaultInstall.exe";
+            process.StartInfo.Verb = "runas";
+            process.StartInfo.UseShellExecute = true;
+            process.StartInfo.CreateNoWindow = true;
+            process.StartInfo.Arguments = $"{filePath}";
+            process.Start();
+            await process.WaitForExitAsync(cts.Token);
```

Comment:
Where is the boolean result going? In case that user cancels the UAC, even if we don't actually read the result of the method, we still need to surround the code in a try-catch block and catches `Win32Exception`.